protobuf/source_reflection_test.go

```diff
@@ -0,0 +1,74 @@
+package protobuf
+
+import (
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+func TestReflection(t *testing.T) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	defer ln.Close()
+	require.NoError(t, err)
+	s := grpc.NewServer()
+	reflection.Register(s)
+	go s.Serve(ln)
+
+	source, err := NewDescriptorProviderReflection(ReflectionArgs{
+		Timeout: time.Second,
+		Peers:   []string{ln.Addr().String()},
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, source)
+
+	result, err := source.FindSymbol("grpc.reflection.v1alpha.ServerReflectionRequest")
```

Comment:
I recall discussing offline something related to the fully-qualified name. Did we confirm that its impossible for their to be multiple types with the same fully qualified name, such as:

```
// foo/foo.proto

package foo;

message foo {
  bar name = 1;
}

message bar {}
```

```
// foo/foo2.proto

package foo.foo;

import "foo/foo.proto"

message bar {}
```

Where `foo.foo.bar` represents both the nested-message in `foo/foo.proto` and the top-level message in `foo/foo2.proto`.

Reply:
Our implementation of the reflection server won't start if there's ambiguity in the fully qualified names that are registered with the server.